File: General/Longest Harmonious Subsequence/LHS.py
from collections import Counter

class Solution:
    def findLHS(self, nums):
        freq = Counter(nums)
        res = 0
        for key in freq:
            if key + 1 in freq:
                res = max(res, freq[key] + freq[key + 1])
        return res

# Example
if __name__ == "__main__":
    nums = [1, 3, 2, 2, 5, 2, 3, 7]
    s = Solution()
    print(s.findLHS(nums))  # Output: 5

File: General/Longest Harmonious Subsequence/LHS.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int findLHS(vector<int>& nums) {
        unordered_map<int, int> freq;
        for (int num : nums) freq[num]++;
        int maxLen = 0;
        for (auto& [num, count] : freq) {
            if (freq.count(num + 1)) {
                maxLen = max(maxLen, count + freq[num + 1]);
            }
        }
        return maxLen;
    }
};

int main() {
    vector<int> nums = {1,3,2,2,5,2,3,7};
    Solution s1;
    cout << s1.findLHS(nums) << endl;
    return 0;
}
